posthog/temporal/data_modeling/run_workflow.py

```diff
@@ -727,6 +728,14 @@ async def run(self, inputs: RunWorkflowInputs) -> Results:
         )
         completed, failed, ancestor_failed = results
 
+        # publish metrics
+        if ancestor_failed:
+            get_data_modeling_finished_metric(status="ancestor_failed").add(1)
+        elif failed:
+            get_data_modeling_finished_metric(status="failed").add(1)
+        elif completed:
+            get_data_modeling_finished_metric(status="completed").add(1)
+
```

Comment:
Hmm the problem is that all 3 sets could have values, I'm wondering if we instead want to do something like:
```
get_data_modeling_finished_metric(status="ancestor_failed").add(len(ancestor_failed))
```
For example. 

I think the question is whether we are measuring the temporal activity (in which case we should wrap this in a try/except and increment failed when exception is raised), or we are measuring individual model execution. Or something else? What do you think?

Reply:
yes, good question, I'm also not sure exactly. I think the idea was to just add some basic alerting to the materialization workflow so we're aware when things are failing. This was why I chose to increment the counter by 1, so we can check what percentage are failing (if we add `len(ancestor_failed)` this becomes less clear). I can ask whoever added this task to clarify exactly what we want here though

Reply:
so it [sounds like](https://posthog.slack.com/archives/C019RAX2XBN/p1741793449595689?thread_ts=1741793105.126149&cid=C019RAX2XBN) we just want some simple alerting for when a modeling run fails, so I think it's ok as it is but let me know if you disagree

Reply:
Given that we want simpler error alerting, I think it makes sense to combine `ancestor_failed` and `failed` together.

Reply:
The reason for this is that `ancestor_failed` will always have at least one model in `failed` (the "ancestor" that "failed"). So, if we are not tracking individual models, it makes sense to make everything just "failed".

Reply:
I originally imagined more granular tracking, which is why I suggested the split in the first place. I still think that's the way to go, but I'm okay with the simpler step in between.

Reply:
Ok, have reverted back to what it was before. Guess we can always add more details later (eg how many ancestors do modeling jobs have on average, how about the ones which fail, why do they fail, etc). But at least we (will) have alerting for now

Reply:
Thanks, sorry for all the confusion here!